desktop/core/src/desktop/settings.py

```diff
@@ -383,6 +383,11 @@
     'default': {
         'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
         'LOCATION': 'unique-hue'
+    },
+    'hive_discovery': {
+        'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
```

Comment:
Why not use LocMemCache instead?
(bit simpler in case we hit issues on the /tmp)

Reply:
The design of the locmemcache creates inconsistency on the server

> This is the default cache if another is not specified in your settings file. If you want the speed advantages of in-memory caching but don’t have the capability of running Memcached, consider the local-memory cache backend. This cache is per-process (see below) and thread-safe
  
So what would happen is different threads would have issues with failovers because every cache entry would die at a different time depending on when it got the value.

A secondary benefit is that it opens up the possibility of spreading hue over the available hiveserver2s instead of hard coding it to one.

Reply:
Thanks for the explanation! Maybe add a small comment too?
(not sure if it will help but we are looking at introducing Redis as caching option via https://issues.cloudera.org/browse/HUE-8738)

If we see issues with the introduction of the cache in traditional Hive, we might just disable then automatically depending on the ini configs

Reply:
So redis (or distributed memcached) opens up some possibilities for hiveserver2 balancing as each hue server would register and you can make sure to round robin the hiveserver2s.  it also would help with the thrift session issue when hive is restarted and hue is not (which is currently not trapped but passed to the UI).

Another architectural issue it would help with is downstream sticky sessions.  Since gunicorn can't use them, there are issues with having multiple HTTPFS servers for HA (which currently have to be resolved via HA Proxy on the HTTPFS side increasing).

Reply:
Thanks again for the detail, and we could work on this after. Realistically the Task Server with Celery should be in v1 in a few weeks, and we should be able to piggyback on the Redis for some of the use cases so.